ccl/multiregionccl: TestShowCreateTable failed [multiregion features surprisingly disabled in shared process]

ccl/multiregionccl.TestShowCreateTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/13104083?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/13104083?buildTab=artifacts#/) on release-23.2.0-rc @ [2a358ad1a84e6ac6ed18828f20f6bd3b98a12a6a](https://github.com/cockroachdb/cockroach/commits/2a358ad1a84e6ac6ed18828f20f6bd3b98a12a6a):


```
=== RUN   TestShowCreateTable
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/9835a78d5c6d74faef4e1b21e8cf50d9/logTestShowCreateTable4206069016
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:159: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    show_create_table.go:75: pq: setting sql.virtual_cluster.feature_access.multiregion.enabled disallows use of multi-region abstractions
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9835a78d5c6d74faef4e1b21e8cf50d9/logTestShowCreateTable4206069016
--- FAIL: TestShowCreateTable (4.14s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowCreateTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34546

```
=== RUN   TestSettingPlacementAmidstAddDrop
I231212 18:46:05.802175 1 (gostd) rand.go:250  [-] 1  random seed: -1746146760369469477
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/9835a78d5c6d74faef4e1b21e8cf50d9/logTestSettingPlacementAmidstAddDrop2758696843
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestSettingPlacementAmidstAddDrop/add_then_set_restricted
=== RUN   TestSettingPlacementAmidstAddDrop/drop_then_set_restricted
=== NAME  TestSettingPlacementAmidstAddDrop
    region_test.go:510: -- test log scope end --
--- PASS: TestSettingPlacementAmidstAddDrop (31.45s)
    --- PASS: TestSettingPlacementAmidstAddDrop/add_then_set_restricted (15.97s)
    --- PASS: TestSettingPlacementAmidstAddDrop/drop_then_set_restricted (15.43s)
=== RUN   TestShowCreateTable
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/9835a78d5c6d74faef4e1b21e8cf50d9/logTestShowCreateTable4206069016
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:159: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    show_create_table.go:75: pq: setting sql.virtual_cluster.feature_access.multiregion.enabled disallows use of multi-region abstractions
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9835a78d5c6d74faef4e1b21e8cf50d9/logTestShowCreateTable4206069016
--- FAIL: TestShowCreateTable (4.14s)
FAIL
```

there's no clear reason why the test failed.

in CRDB logs i see logs indicating "node unavailable" almost from the moment the server starts up:
```
W231212 18:46:41.334646 45154 kv/kvserver/intentresolver/intent_resolver.go:856 ⋮ [-] 310  failed to gc transaction record: could not GC completed transaction anchored at /Table/50/1/‹10›/‹"sql.multi_region.allow_abstractions_for_secondary_tenants.enabled"›: node unavailable; try another peer

W231212 18:46:41.339454 41641 sql/sqlliveness/slinstance/slinstance.go:321 ⋮ [T1,Vsystem,n1] 318  exiting heartbeat loop with error: node unavailable; try another peer
E231212 18:46:41.339470 41641 server/server_sql.go:532 ⋮ [T1,Vsystem,n1] 319  failed to run update of instance with new session ID: node unavailable; try another peer

E231212 18:46:41.343449 44216 jobs/registry.go:992 ⋮ [T10,Vtest-tenant,n1] 331  error getting live session: node unavailable; try another peer
W231212 18:46:41.348228 43139 jobs/adopt.go:521 ⋮ [T1,Vsystem,n1] 332  could not clear job claim: ‹clear-job-claim: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Table/15/1/102,/Table/15/1/103), [txn: 790ae14a], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 32100] RPC error: node unavailable; try another peer›
W231212 18:46:41.351247 44866 jobs/adopt.go:521 ⋮ [T10,Vtest-tenant,n1] 333  could not clear job claim: ‹clear-job-claim: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Tenant/10/Table/15/1/102,/Tenant/10/Table/15/1/103), [txn: b75b936f], [can-forward-ts], [max_span_request_keys: 0], [target_bytes: 32100] RPC error: node unavailable; try another peer›
```

Asking the KV team for help - is this just some kind of overload/networking problem?

--- a/pkg/server/testserver.go
+++ b/pkg/server/testserver.go
@@ -1368,6 +1368,10 @@ func (ts *testServer) StartSharedProcessTenant(
 		}
 	}
 
+	if err = ts.grantDefaultTenantCapabilities(ctx, tenantID, args.SkipTenantCheck); err != nil {
+		return nil, nil, err
+	}
+
 	// Also mark it for shared-process execution.
 	err = execSQL(
 		"start-tenant-shared-service",
@@ -1408,10 +1412,6 @@ func (ts *testServer) StartSharedProcessTenant(
 		pgL:            sqlServerWrapper.loopbackPgL,
 		httpTestServer: hts,
 		drain:          sqlServerWrapper.drainServer,
-	}
-
-	if err = ts.grantDefaultTenantCapabilities(ctx, tenantID, args.SkipTenantCheck); err != nil {
-		return nil, nil, err
 	}
 
 	sqlDB, err := ts.SQLConnE(serverutils.DBName("cluster:" + string(args.TenantName) + "/" + args.UseDatabase))

